### ISSUE TITLE
Need to make `spawn` exit even if there are remainng items in the event loop.

### ISSUE THREAD

**Issue by mikemaccana** [arc-repos/architect#358]
**Describe the bug**
Found as part of #354.

In a real AWS Lambda, [`context.callbackWaitsForEmptyEventLoop = false;`](https://docs.aws.amazon.com/lambda/latest/dg/nodejs-prog-model-context.html) can be set - this allows one to run a lambda without [disconnecting from database](https://www.mongodb.com/blog/post/optimizing-aws-lambda-performance-with-mongodb-atlas-and-nodejs). This is used to keep Lambda connections open on MongoDB, RethinkDB, etc.

Arc needs to simulate AWS's behavior locally. 

**We need to create the equivalent of `callbackWaitsForEmptyEventLoop` = false (ie, exit as soon as all processing is done, rather than waiting for the event loop to be empty?**

**To Reproduce**

Here is some demo code. Change `FUNCTION_NAME` to `'connectOnly'` or `'connectAndDisconnect'` to demonstrate the behavior.

Right now `'connectAndDisconnect'` works, but `'connectOnly'` does not - it finishes processing, but won't ever exit node since the connection is still open.

```
let spawn = require('child_process').spawn
var log = console.log.bind(console)
let minify = script=> '"' + script.replace(/\n/g, '').trim() + '"'

var functions = {
    connectAndDisconnect: `const rethinkdb = require('rethinkdb');    
    ;(async function(request, context) { 
        var connection = await rethinkdb.connect({ host: 'localhost', port: 28015 }); 
        await connection.close();
        console.log('I connected successfully!');
    })();`,
    connectOnly: `const rethinkdb = require('rethinkdb');    
    ;(async function(request, context) { 
        var connection = await rethinkdb.connect({ host: 'localhost', port: 28015 }); 
        console.log('I connected successfully!');
    })();`
}

const FUNCTION_NAME = 'connectOnly'

log(`Testing ${FUNCTION_NAME}`)

var functionToTest = functions[FUNCTION_NAME]

var child = spawn('node', ['-e', minify(functionToTest)], { 
    shell: true,
    stdio: [0, 'pipe', 'pipe'],
    cwd: 'C:\\Users\\mikem\\OneDrive\\Documents\\appless\\documentdb-arc-test\\src\\http\\get-index',
    env: {}
})

var completed = false;

let timeout = 5000;
// bake a timeout
setTimeout(function () {
    log(`Timed out after ${timeout} ms!`)
    child.kill()
}, timeout)

// 'close' will occur if the process finishes naturally
// 'exit' will occur if the process is killed
console.log(`What we want is to be see a 'close' event using the 'connectOnly' code above`)
;['close', 'exit'].forEach(function(event){
    child.on(event, function () {
        log(`'${event}' event happened!`)
        if ( ! completed ) {
            completed = true
            log(`Done!`)
        }
    })
})
```

**What happens?**

Unless we close the connection, the lambda will never complete and will time out

**Expected behavior**

Once the lambda has completed processing it should exit, even if the connection is still open.

**Comment by mikemaccana** [comment]
Alternatively `lambda-local` simply uses `require` to load the lambda, instead of `spawn`. The lamdas execute normally, run their callback and will be considered to be finished even if there are open connections, unlike with `spawn` where the process never terminates:

https://github.com/ashiina/lambda-local/blob/develop/lib/lambdalocal.js#L180

**Comment by brianleroux** [comment]
We used to do that but won't fly w other runrltimes unfortunately

**Comment by mikemaccana** [comment]
Yeah I notice it's used to launch Python. I hate having two code paths, but I don't see any way to make a spawned node process exit when compution is done, even if there are handles open. 

Unless: we spawn a node process that does the require... 🤔

**Comment by mikemaccana** [comment]
I'm gonna hack here. BRB.

**Comment by brianleroux** [comment]
Cool, worth exploring! God speed (also don't mind a bit of divergent paths as long as semantics same)

**Comment by mikemaccana** [comment]
Just a quick update that I've been working on other things and haven't had the time to create a proper fix. If someone else wants to grab it be my guest,
